From 33a5933dc186d20ead3b4b13e1e2eee275e4c427 Mon Sep 17 00:00:00 2001
From: Cheick TRAORE <cheick.traore-ext@st.com>
Date: Thu, 20 Jun 2024 10:52:28 +0200
Subject: [PATCH 0986/1695] drivers: stm32_uart: Make UART init from dt only
 during probe

Remove the function stm32_uart_init_from_dt_node() from the driver
since it is no more used.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 8b9a00a064441d9c206549c00c0b2db487b5b955 drivers: stm32_uart: Make UART init from dt only during probe]
Change-Id: I5cd95a1b6f3448def94781330e73dae68a94bf9e
Signed-off-by: Cheick TRAORE <cheick.traore-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/387883
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/stm32_uart.c         | 21 ---------------------
 core/include/drivers/stm32_uart.h | 13 -------------
 2 files changed, 34 deletions(-)

diff --git a/core/drivers/stm32_uart.c b/core/drivers/stm32_uart.c
index 421f81680..55b847e13 100644
--- a/core/drivers/stm32_uart.c
+++ b/core/drivers/stm32_uart.c
@@ -252,27 +252,6 @@ static void setup_resources(struct stm32_uart_pdata *pd)
 		register_non_secure_uart(pd);
 }
 
-struct stm32_uart_pdata *stm32_uart_init_from_dt_node(void *fdt, int node)
-{
-	struct stm32_uart_pdata *pd = NULL;
-
-	if (fdt_get_status(fdt, node) == DT_STATUS_DISABLED)
-		return NULL;
-
-	pd = calloc(1, sizeof(*pd));
-	if (!pd)
-		panic();
-
-	if (parse_dt(fdt, node, pd)) {
-		/* stm32_uart_init_from_dt_node() expects resources are ready */
-		panic();
-	}
-
-	setup_resources(pd);
-
-	return pd;
-}
-
 static bool uart_is_for_console(void *fdt, int node, char **params)
 {
 	static int uart_console_node = -1;
diff --git a/core/include/drivers/stm32_uart.h b/core/include/drivers/stm32_uart.h
index 6054df72c..0b1783c53 100644
--- a/core/include/drivers/stm32_uart.h
+++ b/core/include/drivers/stm32_uart.h
@@ -29,17 +29,4 @@ struct stm32_uart_pdata {
  */
 void stm32_uart_init(struct stm32_uart_pdata *pd, vaddr_t base);
 
-/*
- * stm32_uart_init_from_dt_node - Initialize a UART instance from a DTB node
- * @fdt: DTB base address
- * @node: Target node offset in the DTB
- * Returns an alloced (malloc) and inited UART platform data on success or NULL
- *
- * This function gets a STM32 UART configuration directives from a DTB node
- * and initializes a UART driver instance.
- * When the DTB specifies that the device is disabled, the function returns
- * NULL. Other issues panic the sequence.
- */
-struct stm32_uart_pdata *stm32_uart_init_from_dt_node(void *fdt, int node);
-
 #endif /*__DRIVERS_STM32_UART_H*/
-- 
2.47.2

